src/gui/qgsadvanceddigitizingtools.cpp

```diff
@@ -297,3 +316,23 @@ void QgsAdvancedDigitizingCirclesIntersectionTool::paint( QPainter *painter )
 
   painter->restore();
 }
+
+bool QgsAdvancedDigitizingCirclesIntersectionTool::eventFilter( QObject *obj, QEvent *event )
+{
+  if ( event->type() == QEvent::ShortcutOverride || event->type() == QEvent::KeyPress )
+  {
+    if ( QKeyEvent *keyEvent = dynamic_cast<QKeyEvent *>( event ) )
+    {
+      if ( keyEvent->key() == Qt::Key_Escape )
+      {
+        if ( mToolWidget )
+        {
+          mToolWidget->deleteLater();
+        }
```

Comment:
Is this actually needed, since it's handled in the destructor?

Reply:
@uclaros , not anymore no, removed.